server/src/main/java/org/elasticsearch/search/SearchUsageException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * A subclass of SearchException that indicates a user error when constructing a search
+ * request. These will always return HTTP status code 400 (BAD REQUEST).
+ */
+public class SearchUsageException extends SearchException {
```

Comment:
Would it make sense to call it IllegalArgumentSearchException perhaps?

Reply:
Yeah, that works. Changed in next push.